commit bf0b0b1d33fe82d66e8312d8c2f2b9995f889f66
Author: Marco Martin <notmart@gmail.com>
Date:   Wed Feb 19 14:00:48 2014 +0100

    center x11 icons
    
    xembed icons can'r really be resized, at least center them

diff --git a/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.cpp b/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.cpp
index 1baa23d..6b148b3 100644
--- a/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.cpp
+++ b/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.cpp
@@ -60,7 +60,7 @@ FdoGraphicsWidget::FdoGraphicsWidget(WId winId, QGraphicsWidget *parent)
     d->winId = winId;
 
     setMinimumSize(22, 22);
-    setMaximumSize(22, 22);
+    setMaximumSize(48, 48);
     resize(22, 22);
 
     setCacheMode(QGraphicsItem::NoCache);
@@ -149,8 +149,9 @@ void FdoGraphicsWidget::setupXEmbedDelegate()
 
     X11EmbedDelegate *widget = new X11EmbedDelegate();
     widget->setMinimumSize(22, 22);
-    widget->setMaximumSize(22, 22);
-    widget->resize(22, 22);
+    widget->setMaximumSize(48, 48);
+    widget->resize(size().toSize());
+    widget->move(QPoint(size().width()/2, size().height()/2) - QPoint(11, 11));
 
     connect(widget->container(), SIGNAL(clientIsEmbedded()),
             this, SLOT(handleClientEmbedded()));
@@ -200,6 +201,14 @@ void FdoGraphicsWidget::handleClientError(QX11EmbedContainer::Error error)
     emit clientClosed();
 }
 
+void FdoGraphicsWidget::resizeEvent(QGraphicsSceneResizeEvent *event)
+{
+    if (d->widget) {
+        d->widget.data()->resize(size().toSize());
+    }
+    
+}
+
 }
 
 #include "fdographicswidget.moc"
diff --git a/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.h b/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.h
index f0be2af..fbe9572 100644
--- a/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.h
+++ b/plasma/generic/applets/systemtray/protocols/fdo/fdographicswidget.h
@@ -45,6 +45,7 @@ signals:
 protected:
     void showEvent(QShowEvent *event);
     void hideEvent(QHideEvent *event);
+    void resizeEvent(QGraphicsSceneResizeEvent *event);
 
 private slots:
     void setupXEmbedDelegate();
diff --git a/plasma/generic/applets/systemtray/protocols/fdo/x11embeddelegate.cpp b/plasma/generic/applets/systemtray/protocols/fdo/x11embeddelegate.cpp
index 035541d..7a3715b 100644
--- a/plasma/generic/applets/systemtray/protocols/fdo/x11embeddelegate.cpp
+++ b/plasma/generic/applets/systemtray/protocols/fdo/x11embeddelegate.cpp
@@ -24,7 +24,6 @@
 
 #include <QtCore/QEvent>
 
-
 namespace SystemTray
 {
 
